File: infra-market-server-go/internal/dto/role.go
package dto

// RoleDto 角色信息DTO
type RoleDto struct {
	ID            uint64   `json:"id"`
	Name          string   `json:"name"`
	Code          string   `json:"code"`
	Description   *string  `json:"description"`
	Status        string   `json:"status"`
	PermissionIds []uint64 `json:"permissionIds"`
	CreateTime    string   `json:"createTime"`
	UpdateTime    string   `json:"updateTime"`
}

// RoleFormDto 角色创建/更新表单
type RoleFormDto struct {
	Name          string   `json:"name" binding:"required,min=2,max=50"`
	Code          string   `json:"code" binding:"required,min=2,max=100"`
	Description   *string  `json:"description" binding:"omitempty,max=200"`
	PermissionIds []uint64 `json:"permissionIds" binding:"required,min=1"`
}

// RoleQueryDto 角色查询DTO
type RoleQueryDto struct {
	Name    *string `form:"name"`
	Code    *string `form:"code"`
	Status  *string `form:"status" binding:"omitempty,oneof=active inactive"`
	Current int     `form:"current" binding:"omitempty,min=1" default:"1"`
	Size    int     `form:"size" binding:"omitempty,min=1,max=1000" default:"10"`
}

File: infra-market-server-dotnet/DTOs/Role.cs
using System.ComponentModel.DataAnnotations;

namespace InfraMarket.Server.DTOs;

public class RoleDto
{
    public ulong Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Code { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Status { get; set; } = string.Empty;
    public List<ulong> PermissionIds { get; set; } = new();
    public string CreateTime { get; set; } = string.Empty;
    public string UpdateTime { get; set; } = string.Empty;
}

public class RoleFormDto
{
    [Required]
    [MinLength(2)]
    [MaxLength(50)]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MinLength(2)]
    [MaxLength(100)]
    public string Code { get; set; } = string.Empty;

    [MaxLength(200)]
    public string? Description { get; set; }

    [Required]
    [MinLength(1)]
    public List<ulong> PermissionIds { get; set; } = new();
}

public class RoleQueryDto
{
    public string? Name { get; set; }
    public string? Code { get; set; }
    public string? Status { get; set; }
    public int? Page { get; set; }
    public int? Size { get; set; }
}
